core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -45,6 +45,51 @@ trait EntityDecoder[F[_], T] { self =>
       override def consumes: Set[MediaRange] = self.consumes
     }
 
+  def handleError(f: DecodeFailure => T)(implicit F: Functor[F]): EntityDecoder[F, T] = transform {
+    case r @ Right(_) => r
+    case Left(e) => Right(f(e))
+  }
+
+  def handleErrorWith(f: DecodeFailure => DecodeResult[F, T])(
+      implicit F: Monad[F]): EntityDecoder[F, T] = transformWith {
+    case Right(r) => DecodeResult.success(r)
+    case Left(e) => f(e)
+  }
+
+  def transform[T2](s: T => T2, f: DecodeFailure => DecodeFailure)(
```

Comment:
Would it be appropriate to call this `bimap` and avoid the overload?  Perhaps that implies a `Bifunctor`, which it's not.  I wish this had a different name from the other `transform`, which seems to fit nicely with what cats uses in `OptionT` and `EitherT`. :thinking:

In any event, I think it makes sense to flip `s` and `t`.  That's how `bimap` works, and that's the closest cousin I can think of for this method in the ecosystem.  It's not a big deal because it doesn't compile if the arguments are flipped, but I think it's nice for consistency.

Reply:
I have to laugh because I thought of the exact same thing today :) Renamed those overloads to `bimap` and `biflatMap` respectively and flipped `s` and `f` :)